source/extensions/common/wasm/wasm.cc

```diff
@@ -1324,16 +1329,24 @@ Wasm::Wasm(absl::string_view vm, absl::string_view id, absl::string_view initial
 }
 
 void Wasm::registerCallbacks() {
-#define _REGISTER(_fn)                                                                             \
+#define _REGISTER_ABI(_fn, _abi)                                                                   \
   wasm_vm_->registerCallback(                                                                      \
-      "envoy", #_fn, &_fn##Handler,                                                                \
-      &ConvertFunctionWordToUint32<decltype(_fn##Handler),                                         \
-                                   _fn##Handler>::convertFunctionWordToUint32);
+      "envoy", #_fn, &_fn##_abi##Handler,                                                          \
+      &ConvertFunctionWordToUint32<decltype(_fn##_abi##Handler),                                   \
+                                   _fn##_abi##Handler>::convertFunctionWordToUint32)
+#define _REGISTER(_fn) _REGISTER_ABI(_fn, )
+
   if (is_emscripten_) {
+    if (emscripten_abi_major_version_ > 0 || emscripten_abi_minor_version_ > 1) {
```

Comment:
Do you want to check the precise version and then fail to load the module if it isn't one of the know versions?

Reply:
I don't, because then we'd stop loading working WASM modules (e.g. neither change to ABI 0.1 nor 0.3 required any changes in Envoy-WASM)... unless you feel strongly about it?

Reply:
I'm OK with it this way.